libr/cons/dietline.c

```diff
@@ -186,15 +186,44 @@ static void unix_word_rubout(void) {
 }
 
 static int inithist(void) {
+	if (I.history.data) {
+		RCons *cons = r_cons_singleton ();
+		RCore *core = (RCore *)(cons->user);
```

Comment:
rcons shouldnt be able to access RCore. its rcore who uses rcons. so do this change in core instead to configure the dietline by adding a new api to specify the new history size instead